public_html/wp-content/themes/pattern-directory/src/store/resolvers.js

```diff
@@ -16,16 +17,35 @@ import {
 	loadPatternFlagReasons,
 	loadPatterns,
 } from './actions';
-import { getQueryString } from './utils';
+import { PER_PAGE, getQueryString } from './utils';
+
+async function parseResponse( response ) {
+	try {
+		return {
+			total: response.headers?.get( 'X-WP-Total' ),
+			totalPages: response.headers?.get( 'X-WP-TotalPages' ),
+			results: await response.json(),
+		};
+	} catch ( error ) {
+		return {};
+	}
+}
 
 export function* getPatternsByQuery( query ) {
 	const queryString = getQueryString( query );
 	try {
 		yield fetchPatterns( queryString );
-		const results = yield apiFetch( {
-			path: addQueryArgs( '/wp/v2/wporg-pattern', query ),
+		const response = yield apiFetch( {
+			path: addQueryArgs( '/wp/v2/wporg-pattern', { ...query, per_page: PER_PAGE } ),
+			parse: false,
+		} );
+		const { total, totalPages, results } = yield __unstableAwaitPromise( parseResponse( response ) );
```

Comment:
I don't see this PR making use of the `AWAIT_PROMISE` dispatched action. Is it used internally somewhere or is there a future plan to rely on it for loading the grid?

Reply:
It's just an internal, there's no way to get the results of a promise otherwise.